test/test_charm.py

```diff
@@ -25,53 +23,18 @@
 
 import ops
 import ops.charm
-from ops.model import ModelError, _ModelBackend
-from ops.storage import SQLiteStorage
+from ops.model import ModelError
 
-from .test_helpers import fake_script, fake_script_calls
+from .test_helpers import FakeScript, create_framework
 
 
-class TestCharm(unittest.TestCase):
+@pytest.fixture
+def fake_script_fixture(monkeypatch: pytest.MonkeyPatch, tmp_path: pathlib.Path):
+    return FakeScript(monkeypatch, tmp_path)
 
-    def setUp(self):
-        def restore_env(env: typing.Dict[str, str]):
-            os.environ.clear()
-            os.environ.update(env)
-        self.addCleanup(restore_env, os.environ.copy())
-
-        os.environ['PATH'] = os.pathsep.join([
-            str(Path(__file__).parent / 'bin'),
-            os.environ['PATH']])
-        os.environ['JUJU_UNIT_NAME'] = 'local/0'
-
-        self.tmpdir = Path(tempfile.mkdtemp())
-        self.addCleanup(shutil.rmtree, str(self.tmpdir))
-        self.meta = ops.CharmMeta()
-
-        class CustomEvent(ops.EventBase):
-            pass
-
-        class TestCharmEvents(ops.CharmEvents):
-            custom = ops.EventSource(CustomEvent)
-
-        # Relations events are defined dynamically and modify the class attributes.
-        # We use a subclass temporarily to prevent these side effects from leaking.
-        ops.CharmBase.on = TestCharmEvents()  # type: ignore
-
-        def cleanup():
-            ops.CharmBase.on = ops.CharmEvents()  # type: ignore
-        self.addCleanup(cleanup)
-
-    def create_framework(self):
-        model = ops.Model(self.meta, _ModelBackend('local/0'))
-        # we can pass foo_event as event_name because we're not actually testing dispatch
-        framework = ops.Framework(SQLiteStorage(':memory:'), self.tmpdir, self.meta,
-                                  model)
-        self.addCleanup(framework.close)
-        return framework
-
-    def test_basic(self):
 
+class TestCharm:
+    def test_basic(self, monkeypatch: pytest.MonkeyPatch, request: pytest.FixtureRequest):
```

Comment:
Are you planning to move these to the top level too? There's also `TestMeta` in this file, but those functions could be renamed slightly, for example `test_links` -> `test_meta_links`.

Reply:
Makes sense, this file is mainly one big test class with a smaller one.